app/uk/gov/hmrc/helptosave/repo/EnrolmentStore.scala

```diff
@@ -96,12 +99,18 @@ class MongoEnrolmentStore @Inject() (mongo:   ReactiveMongoComponent,
     )
   )
 
+  private[repo] def doInsert(nino:              NINO,
+                             eligibilityReason: Option[Int],
+                             channel:           String,
+                             itmpFlag:          Boolean)(implicit ec: ExecutionContext): Future[WriteResult] =
+    collection.insert(BSONDocument("nino" -> nino, "itmpHtSFlag" -> itmpFlag, "eligibilityReason" -> eligibilityReason, "channel" -> channel))
+
   private[repo] def doUpdate(nino: NINO, itmpFlag: Boolean)(implicit ec: ExecutionContext): Future[Option[EnrolmentData]] =
     collection.findAndUpdate(
       BSONDocument("nino" -> nino),
       BSONDocument("$set" -> BSONDocument("itmpHtSFlag" -> itmpFlag)),
       fetchNewObject = true,
-      upsert         = true
+      upsert         = false
```

Comment:
do we want `upsert=false`. The description for `upsert` is:
```
If set to true, creates a new document when no document matches the query
criteria. The default value is false, which does not insert a new document when no match is found.
```
In all cases when we're updating I think we'd expect a matching document to be present.

Reply:
yes, thats why i didn't want to make it to `true` because incase if the document is not found for some reason then `true` makes it silently insert new record, but if we use `false` then it fails and we will get a chance to investigate the error 
And also we not making use of the returned record in the `true` case , then i don't see a point to to use `true` flag there

Reply:
you're right - that makes sense